### PR TITLE
glusterd2 unit test cases for gdctx,middleware, pmap and volgen

### PR DESCRIPTION
**Pull request by Madhu-1**
unit test cases for gdctx, middleware,pmap and volgen packages
Signed-off-by: Madhu Rajanna <mrajanna@redhat.com>

### CODE REVIEW

**Comment by obnoxxx** [comment]
@Madhu-1 kudos for adding such a lot of test code!!
My personal style suggestion would be to have 1 commit per new test file here instead of 1 for all the 9 files. That makes each commit a little more isolated and focused. Not sure what @kshlm / @prashanthpai and others think.

File: glusterd2/gdctx/context_test.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	reqID := uuid.NewRandom()
+
+	ctx = WithReqID(ctx, reqID)
+	id := GetReqID(ctx)
+	assert.Equal(t, id, reqID)
+}
+
+func TestWithReqLogger(t *testing.T) {
+	var ctx context.Context
+	lo := log.WithField("test", "test")
```
[the remark this answers is not in the document]
Comment (obnoxxx) [reply glusterd2/gdctx/context_test.go]: agree. this increases readability!

But what is it with the two identical assert lines here? (`assert.NotNil(t, ctx)`) I guess one can just be removed ...

**Comment by prashanthpai** [comment]
> My personal style suggestion would be to have 1 commit per new test file here instead of 1 for all the 9 files. That makes each commit a little more isolated and focused. Not sure what @kshlm / @prashanthpai and others think.

My personal preference is to have one commit per logically-related and fix self-contained fix. It might be across multiple files or packages. I hate to see git log like this:

* Fix typo in pkg/utils.go
* Fix typo in pkg/api.go

In the above example, you can combine all those commits into one, as long as they're typo fixes i.e same type of fix.

**Comment by Madhu-1** [comment]
@obnoxxx , added 1 commit per new test file

@kshlm, @obnoxxx  addressed the review comments.

File: glusterd2/gdctx/context_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Madhu-1) [reply glusterd2/gdctx/context_test.go]: removed extra assert and grouped function calls across all test files.

**Comment by prashanthpai** [comment]
I really don't see any good reason for this to be not a single commit. It doesn't help reviewers in any way and clutters git log output. This is the perfect example of commits that I don't prefer to see ;)
```
unit test cases for gdctx/context.go  …
unit test cases for gdctx/global.go  …
unit test cases for gdctx/id.go  …
unit test cases for middleware/auth.go  …
unit test cases for middleware/recover.go  …
unit test cases for middleware/request_id.go  …
unit test cases for middleware/request_logging.go  …
unit test cases for pmap/pmap.go  …
unit test cases for pmap/rpc_prog.go  …
```

The logical "fix" is that you're adding unit tests. One commit should suffice. If the unit tests are exhaustive for files in a particular package, then one commit per package is also okay.

 This is just my personal preference. Your code will be reviewed and merged regardless ;)

**Pull request merged**